spec/models/rbac_spec.rb

```diff
@@ -17,6 +17,58 @@
     @user  = FactoryGirl.create(:user, :miq_groups => [@group])
   end
 
+  context "tenant scoping" do
+    before do
+      default_tenant = Tenant.seed
+
+      @owner_tenant  = FactoryGirl.create(:tenant, :divisible => false, :parent => default_tenant)
+      @owner_group   = FactoryGirl.create(:miq_group, :tenant => @owner_tenant)
+      @owner_user    = FactoryGirl.create(:user, :userid => 'foo', :miq_groups => [@owner_group])
+
+      @other_tenant  = FactoryGirl.create(:tenant, :divisible => false, :parent => default_tenant)
+      @other_group   = FactoryGirl.create(:miq_group, :tenant => @other_tenant)
+      @other_user    = FactoryGirl.create(:user, :userid => 'bar', :miq_groups => [@other_group])
+
+      @owner_vm      = FactoryGirl.create(:vm_vmware, :tenant => @owner_tenant)
+    end
+
+    it ".search with :userid, finds user's tenant vms" do
+      results, = Rbac.search(:class => "Vm", :results_format => :objects, :userid => @owner_user.userid)
+      expect(results).to eq [@owner_vm]
+    end
+
+
+    it ".search with :userid filters out other tenants" do
+      results, = Rbac.search(:class => "Vm", :results_format => :objects, :userid => @other_user.userid)
+      expect(results).to eq []
+    end
+
+    it ".search with User.with_userid finds user's tenant vms" do
+      User.with_userid(@owner_user.userid) do
+        results, = Rbac.search(:class => "Vm", :results_format => :objects)
+        expect(results).to eq [@owner_vm]
+      end
+    end
+
+    it ".search with User.with_userid filters out other tenants" do
+      User.with_userid(@other_user.userid) do
```

Comment:
@kbrock @gtanzillo Is this going away or better said, should I have test cases that test `with_userid` blocks, explicit  `:userid` and/or `:miq_group_id` passed to `.search` and does it makes sense to mix them even in conflicting situations?

For example:
* Owner group, other user (not in owner group):
`Rbac.search(:class => "Vm", :results_format => :objects, :miq_group_id => @owner_group.id, :userid => @other_user.userid)`

* Other group, owner user (not in other group):
`Rbac.search(:class => "Vm", :results_format => :objects, :miq_group_id => @other_group.id, :userid => @owner_user.userid)`

* Owner group, with_userid block for conflicting user:
```ruby
User.with_userid(@owner_user.userid) do
  Rbac.search(:class => "Vm", :results_format => :objects, :userid => @other_user.userid)
end
```